docker-compose.yaml

```diff
@@ -4,22 +4,21 @@ services:
   ffc-demo-web:
     build: .
     image: ffc-demo-web
-    command: npm run start:watch
     depends_on:
       - redis
     environment:
       API_GATEWAY: http://ffc-demo-api-gateway:3001
       COOKIE_PASSWORD: cookiecookiecookiecookiecookiecookie
       REDIS_HOSTNAME: redis
     volumes:
-      - node_modules:/usr/src/app/node_modules
-      - .:/usr/src/app
+      - ./app/:/home/node/app/
+      - ./package.json:/home/node/package.json
+      - ./package-lock.json:/home/node/package-lock.json
 
   redis:
     image: redis
     volumes:
       - redis_data:/data
 
 volumes:
-  node_modules:
```

Comment:
The `node_modules` volume is a common trick to exclude a sub-folder from a bind-mount. However, it has caused confusion for several developers and was found by one of the docker-compose developers to [rely on a race-condition](https://github.com/docker/compose/issues/4337#issuecomment-358459746) in the Docker engine so should not be depended on. It also causes an unexpected `node_modules/` folder to be created on the host, possibly a side-effect of the race condition.

Restricting volume mounts to only the files we want allows this trick to be avoided. Alternatively, we could switch from NPM to Yarn and [put node_modules outside the app root](https://nickjanetakis.com/blog/docker-tip-75-how-to-avoid-node-modules-in-your-volume-mounts) within the container. However, I would still feel that there is a strong benefit to listing all the files/folders we want mounted in `docker-compose.yaml`.